python/tk_multi_workfiles/browser_form.py

```diff
@@ -378,22 +405,74 @@ def _on_selected_entity_changed(self, selection_details, breadcrumb_trail):
                     details.is_leaf = is_leaf
                     search_details.append(details)
 
+            # If we can enable user filtering, show or hide the user filter widget
+            # if one of the entities in the selection has sandboxes.
+            if self._enable_user_filtering:
+                # The elementes in the selection might not always be an entity, e.g.
+                # if the user clicked the "Character" category, so filter None entities.
+                flat_selection = [child_detail["entity"] for child_detail in children if child_detail is not None]
+                if primary_entity:
+                    flat_selection.append(primary_entity)
+
+                # For each form, show the button if its content uses sandboxes.
+                for form in self._file_browser_forms:
+                    if form.work_files_visible and self._uses_sandboxes(flat_selection, workfiles=True):
+                        form.show_user_filtering_widget(True)
+                    elif form.publishes_visible and self._uses_sandboxes(flat_selection, publishes=True):
+                        form.show_user_filtering_widget(True)
+                    else:
+                        form.show_user_filtering_widget(False)
+        else:
+            # Selection is empty, so there surely is no user filtering available.
+            for form in self._file_browser_forms:
+                form.show_user_filtering_widget(False)
+
         # refresh files:
         if self._file_model:
             p_details = []
             for search in search_details:
-                p_details.append({"name":search.name, 
-                       "entity":search.entity, 
-                       "is_leaf":search.is_leaf, 
-                       "child_entities":search.child_entities})
+                p_details.append({
+                    "name":search.name,
+                    "entity":search.entity,
+                    "is_leaf":search.is_leaf,
+                    "child_entities":search.child_entities
+                })
 
             self._file_model.set_entity_searches(search_details)
 
         # emit work-area-changed signal:
         self._emit_work_area_changed(primary_entity or None, breadcrumb_trail)
-        
+
         return primary_entity
 
+    def _uses_sandboxes(self, flat_selection, workfiles=False, publishes=False):
+        """
+        Checks if any items in the selection uses sandboxes.
+
+        :param flat_selection: Array of elements inside the selection.
+        :param workfiles: If True, the method will consider that a sandbox is used
+            if workfiles have user sandboxes.
+        :param publishes: If True, the method will consider that a sandbox is used
+            if publishes have user sandboxes.
+
+        :returns: True if there are user sandboxes for the selection, False otherwise.
+        """
+        # Turn on or off user sandbox filter button.
+        app = sgtk.platform.current_bundle()
```

Comment:
Honestly, it is good practice to not keep your own pointer.  Right now that call always returns that same thing, but what if at a future date it doesn't?  Kind of like keeping a pointer to the shotgun API isn't great since we generate them per thread and it leaves the door open to ending up using the same object across threads.